docs/docs/api/observers.mdx

```diff
@@ -30,6 +30,21 @@ properly.
 
 :::
 
+#### Performance optimizations
+
+Mostly you try to make `Observer` as small as possible. But sometimes you want to exclude
+the widget subtree from being rebuilt and improve the performance.
+
+Real world examples:
+
+1. You have a colored Container with some heavy widgets inside (like ListView with shrinkWrap: true).
+You want to change the color by Observable, but don't want to rebuild the whole ListView.
+2. You have some fancy animation driven by Observable. You want to animate the widget and want to
+preserve its children from being rebuilt.
+
+For that you can use an `Observer.withBuiltChild` constructor. It takes `builderWithChild`
```

Comment:
@pavanpodila @amondnet I'm thinking about making this technique in a more MobX way.  
Observer is already protects its subtree from being rebuit by the parental Observer. But it should observe something.   
Can we come up with some widget like RebuiltProtector? Which does nothing but preserves its subtree. So we can get rid of this awkward `withBuiltChild` constructor.

Reply:
Hi @subzero911,

Are you suggesting more of a refactoring or a new method of doing the rebuilt-child protection ? I see your point about having a separate Widget to do it. I was originally going for a factor based approach so all "Observers" technically are under the Observer category with factory methods to give more refinement.

Reply:
Look, currently inner Observer already preserves its child from the outer Observer's rebuild.
```dart
Observer(
  buider: (_) {
    return Container(  // rebuilds on ctrl.observableColor change
       color: ctrl.observableColor,  
       child: Observer(
         builder: (_) {
           return Text(ctrl.observableText);  // does not rebuild on ctrl.observableColor change
         }
       )
```
My idea is to make a variant of Observer which doesn't observe anything, and its sole purpose is to preserve its child from the outer Observer.

Reply:
Ah ... ok ... Isn't that something you could get from Flutter directly by using the same Widget Type and Key ?

Reply:
> Ah ... ok ... Isn't that something you could get from Flutter directly by using the same Widget Type and Key ?

But I cannot use Observer for that:
1. It constantly prints "there are no observables in builder"
2. It will be unclear to the user, that Observer is used for preserving descendants from rebuilds, not for observing reactive variables.

Reply:
Ok. Perhaps it can be named as a PassThroughObserver that focuses on retaining built child? Is there an observable behaviour needed here?

Reply:
How about making it a bit verbose: `BuiltChildObserver` ?

Reply:
I would like the name `RebuildPreventor`. It may extend `Observer` but a user shouldn't know about it.

Reply:
To my opinion, the most correct way is to add the second optional parameter `child` to the `builder` function.
```dart
builder: (context, child) => ...
```
But it leads to a breaking change. All the users would have to make changes in their Observers. VsCode "wrap to Observer" extension will break as well.
That's why I added the second `.withBuiltChild` constructor.
But it has several drawbacks. I had to change a `builder` function to nullable, and add runtime asserts that either `builder` or `builderWithChild + child` are not null.

Reply:
Ya let's avoid the breaking change ... we can go ahead with the new custom widget ... naming is still not clear to me ...We need to ensure it is intent-driven and clear...Can we explore a few more names before we settle ?

Reply:
`BuiltSubtree`, `BuiltDescendant`, `RebuildProtector`, `RebuildStopper`, `RebuildPreventor`, `RebuildGuard`...
Or maybe "Prebuilt" instead of "Built"

Reply:
`BuiltChildObserver`

Reply:
I don't like the part `Observer`, because it doesn't observe anything.

Reply:
If you're going to extend it from Observer, maybe it should be just `Observer` with `observe: false` parameter?

Reply:
By the way, I checked this twice (by `print`), and Observer does not preserve its child from being rebuilt by the parent Observer.
If the parent Observer got triggered, the descendant Observer rebuild as well.
So the only 2 ways are: 1. using `const`. 2. using prebuilt `child` branch.

Reply:
But if I set up it like this:
<img width="541" alt="image" src="https://github.com/user-attachments/assets/aa350031-25f0-423a-83a3-79f2dae8c87b">

Now it works correctly:
<img width="208" alt="image" src="https://github.com/user-attachments/assets/12275c22-0d68-49e0-bf0e-82bc75a34313">